bioimageio/core/measure_groups.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import warnings
+from itertools import product
+from typing import Dict, Hashable, Iterator, List, Mapping, Optional, Sequence, Tuple, Type, Union
+
+import numpy
+import xarray as xr
+
+from bioimageio.core.statistical_measures import Mean, Measure, Percentile, Std, Var
+from .utils import TensorName
+
+try:
+    import crick
+except ImportError:
+    crick = None
+
+MeasureValue = xr.DataArray
+
+
+class MeasureGroup:
```

Comment:
I added the docstring "group of measures for more efficient computation of multiple measures"
`MeanVarStd(MeasureGroup)` computes Mean, Var, and Std together, and the `...Percentiles(MeasureGroup)` computes multiple percentiles together.  Atm they can only be used for batched computation ('update_with_sample` + `finalize`). I am considering to add either separate groups for the sample case or using the same groups for simplicity (they should usually group just like the dataset case, although other algorithms might be used). 

The `Measure().compute` method could be used for reference (in the tests!) or (with a simple wrapper) as a trivial "group" (with a wrapper, because groups return Dict[Measure, MeasureValue], while `Measure` simply returns `MeasureValue`)

The mapping of a `Dict[TensorName, Set[Measure]]` to `List[MeasureGroup]` is done by `get_measure_groups`, see other comment.